runtime/common/bigint.go

```diff
@@ -42,17 +44,46 @@ func NewBigIntFromAbsoluteValue(gauge MemoryGauge, value *big.Int) *big.Int {
 	return new(big.Int).Abs(value)
 }
 
+const minBigIntLength = 4 * BigIntWordSize
+
 // OverEstimateBigIntFromString is an approximate inverse of `interpreter.OverEstimateBigIntStringLength`.
 // Returns the estimated size in bytes.
-func OverEstimateBigIntFromString(s string) int {
+func OverEstimateBigIntFromString(s string, literalKind IntegerLiteralKind) int {
```

Comment:
Is it possible that the input `s` has leading zeros? If yes, it would be great to work with `l` as the non-zero length instead of `len(s)`.

Reply:
That's a good catch! yes, it is possible. Handled this ind6cc134ad53db865bf3572bb65e60f544d1040a2